[feat](auth) M3 Functional Increase

CavernaryBeast opened the pull request:
This pull-request references the Pull-Request previously done #45 
Summary of changes:

- A custom model has been implemented. It is under test.
The main change is: Users will have more than 1 email for email verification

CavernaryBeast committed:
[feat](authentication) Login and basic SignUp implementation

A basic Login functionality and SignUp has been implemented. Also, thanks to the functionalities Django offers, a reset password is available too.

fhera committed:
[feat](authentication) Login templates

fhera committed:
[refactor](auth)Clean unused libraries

- Clean unused libraries
- Modify gitignore

fhera committed:
[feat](auth) Upload the libraries we will need

fhera committed:
[refactor](auth) Login with Github

- Create index page
- Refactor urls
- Refactor all templates
Closes #26

fraguerrero committed:
Cambios para extender el modelo User con emails de cuentas externas.  Closes #38

fraguerrero committed:
Cambios para extender el formulario de registro con emails asociados a cuentas externas de usuario  Closes #31

Commit:
Merge branch 'auth-feature/extend-email-signup' into auth

pedroswe reviewed (changes requested):
Es necesario arreglar el código para que se ejecuten las builds de travis correctamente.
Para ampliar la información, revisad la issue #51

serortagu reviewed (changes requested):
Se han detectado diversos errores referentes a bibliotecas importadas y CSS provenientes del anterior Milestone. Es necesario arreglarlo pues el sistema no funciona correctamente, para más información consulte las issues #52 y #53.

CavernaryBeast committed:
[fix](auth) Corrección de errores CSS y Tests de Travis (#61)

* Revert "Cambios para extender el formulario de registro con emails asociados a cuentas externas de usuario  Closes #31"

This reverts commit 934de025

* Revert "Cambios para extender el modelo User con emails de cuentas externas.  Closes #38"

This reverts commit 0c24d142

* Cambios para extender el modelo User con emails de cuentas externas.  Closes #38

* Cambios para extender el formulario de registro con emails asociados a cuentas externas de usuario  Closes #31

* Cambios para extender el modelo de User con emails asociados a cuentas externas de usuario

Closes #38

*[fix[(auth) Arreglo de los Test de Travis

Closes #51 

* [fix](auth) Arreglo del error de CSS

Con el objetivo de arreglar el error que provocaba el CSS detallado en el Issue #53 , se han hecho cambios:

En las páginas que extendían de base.html de nuestro módulo.

Se ha creado un nuevo archivo CSS para ser usado en base.html y que no influya en los archivos que extiendan de este.

Co-authored-by: fraguerrero <ramon.guerrero@gmail.com>

fhera committed:
Auth feature/fix issue52 (#60)

* [fix](auth) Removed dependencies downloaded by nodejs

Removed dependencies downloaded by nodejs and restored by other methods

* [fix](auth) modified the loading of libraries to specific templates

* [fix](auth) modified the loading of libraries to specific templates

Co-authored-by: Raimundo <47127494+CavernaryBeast@users.noreply.github.com>

CavernaryBeast committed:
Merge branch 'develop' into auth

serortagu reviewed (approved):
Se han hecho los cambios pertinentes para hacer que el CSS sea menos intrusivo.

pedroswe reviewed (approved):
Se han realizado los cambios requeridos en la issue #51. Buen trabajo @CavernaryBeast  @fhera

CavernaryBeast committed:
[fix](auth)Corrección final de errores provocados por CSS (#62)

Con este commit queda cerrada la Issue #53.

CavernaryBeast commented:
### Modelo de usuario actualizado -> Posibles errores

Para aquellos equipos que no hayan estado trabajando con nuestro subsistema actualizado podrían surgir pequeños problemas con el nuevo modelo de usuario. Para ello, debemos seguir los siguientes pasos para corregir el problema:

1. Borrar la base de datos

2. Borrar la carpeta migrations de la carpeta authentication en decide/authentication

3. Ejecutar ./manage.py makemigrations

4. Ejecutar ./manage.py migrate
